agrona/src/main/java/org/agrona/concurrent/ringbuffer/RingBuffer.java

```diff
@@ -21,6 +21,25 @@
 /**
  * Ring-buffer for the concurrent exchanging of binary encoded messages from producer(s) to consumer(s)
  * in a FIFO manner.
+ * Example usage:
+ * <pre>
+ * // Create a ring buffer with a direct buffer of 1024 bytes
+ * ByteBuffer byteBuffer = ByteBuffer.allocateDirect(1024);
+ * RingBuffer ringBuffer = new OneToOneRingBuffer(new UnsafeBuffer(byteBuffer));
+ *
+ * // Write a simple message
+ * int length = 8;
+ * MutableDirectBuffer srcBuffer = new UnsafeBuffer(ByteBuffer.allocateDirect(length));
```

Comment:
This code gives the impression it is fine to create ad hoc MutableDirectBuffers. In low latency systems, you don't want to create litter on the fast path because it creates a lot of performance problems. From increased mean to increased latency tail.

It should be clear to the end user that a cache buffer should be used.